pkg/cmd/upgrade/upgrade_boot.go

```diff
@@ -462,3 +472,36 @@ func (o *UpgradeBootOptions) gitProvider(gitInfo *gits.GitRepository) (gits.GitP
 	}
 	return provider, nil
 }
+
+func (o *UpgradeBootOptions) updatePipelineBuilderImage(versionStreamURL string, versionStreamRef string) error {
+	fileName := filepath.Join(o.Dir, config.ProjectConfigFileName)
+	projectConf, err := config.LoadProjectConfigFile(fileName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to load project config file %s", fileName)
+	}
+	image := projectConf.PipelineConfig.Pipelines.PullRequest.Pipeline.Agent.Image
+	builder := image[:strings.IndexByte(image, ':')]
+
+	resolver, err := o.CreateVersionResolver(versionStreamURL, versionStreamRef)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create version resolver for %s", builder)
+	}
+	imageVersion, err := resolver.ResolveDockerImage(builder)
+	if err != nil {
+		return errors.Wrapf(err, "failed to resolve image %s", builder)
+	}
+
+	log.Logger().Infof("Updating pipeline agent images to %s", util.ColorInfo(imageVersion))
+	projectConf.PipelineConfig.Pipelines.PullRequest.Pipeline.Agent.Image = imageVersion
+	projectConf.PipelineConfig.Pipelines.Release.Pipeline.Agent.Image = imageVersion
```

Comment:
wonder if we need to check the individual steps and see if they specify an image and update those too?

Reply:
also I think we can have a pipeline that only specifies a `PullRequest` or a `Release` so I think we could hit a `nil` pointer above?  Not 100% sure but a quick test with a couple of sample `jenkins-x.yml` might be worth it to check?